typing_extensions/src_py3/typing_extensions.py

```diff
@@ -1643,14 +1643,14 @@ def _typeddict_new(*args, total=True, **kwargs):
             raise TypeError("TypedDict takes either a dict or keyword arguments,"
                             " but not both")
 
-        ns = {'__annotations__': dict(fields), '__total__': total}
+        ns = {'__annotations__': dict(fields)}
         try:
             # Setting correct module is necessary to make typed dict classes pickleable.
             ns['__module__'] = sys._getframe(1).f_globals.get('__name__', '__main__')
         except (AttributeError, ValueError):
             pass
 
-        return _TypedDictMeta(typename, (), ns)
+        return _TypedDictMeta(typename, (), ns, total=total)
```

Comment:
On Python 3.4 and 3.5 it seems that `_TypedDictMeta(...)` calls `__init__` rather than `__new__`, but `__init__` isn't implemented. See, for instance: https://travis-ci.com/github/python/typing/jobs/473002561#L338

Reply:
That looks nasty. I wonder if it's related to this note in https://docs.python.org/3/whatsnew/3.6.html#changes-in-the-python-api : 

> As part of PEP 487, the handling of keyword arguments passed to type (other than the metaclass hint, metaclass) is now consistently delegated to object.__init_subclass__(). This means that type.__new__() and type.__init__() both now accept arbitrary keyword arguments, but object.__init_subclass__() (which is called from type.__new__()) will reject them by default. Custom metaclasses accepting additional keyword arguments will need to adjust their calls to type.__new__() (whether direct or via super) accordingly.

I just tried and failed to get pyenv to compile 3.5 for me so I can't try this out myself :( If it's too hard to get this to work I wouldn't be opposed to fixing it for 3.6+ only for now.